src/Maths/src/Box2.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+
+namespace Silk.NET.Maths
+{
+    /// <summary>
+    /// A structure representing a Box2 with a <see cref="Min"/> and <see cref="Max"/>
+    /// </summary>
+    public struct Box2<T>
+        : IEquatable<Box2<T>>
+        where T : unmanaged, IFormattable, IEquatable<T>, IComparable<T>
+    {
+        /// <summary>
+        /// The min.
+        /// </summary>
+        public Vector2<T> Min;
+        /// <summary>
+        /// The max.
+        /// </summary>
+        public Vector2<T> Max;
+        
+        /// <summary>
+        /// Constructs a Box2 from a min and a max
+        /// </summary>
+        /// <param name="min">The min of the rect.</param>
+        /// <param name="max">The max of the rect.</param>
+        public Box2(Vector2<T> min, Vector2<T> max)
+        {
+            Min = min;
+            Max = max;
+        }
+
+        /// <summary>
+        /// The center of this box.
+        /// </summary>
+        public Vector2<T> Center
+        {
+            get => Min + Max / Scalar<T>.Two;
```

Comment:
Can't be a property due to serializers getting thrown off by them.
Please convert to a method, the method version may share the same name for less typing.